src/executor.ts

```diff
@@ -80,45 +93,56 @@ export class Executor extends SmartContract {
    Player flips coin
    Smart contract verifies randomness from Oracle
   */
-
-  // TODO: add randomness element(s)
   @method
   flipCoin(
     player: PublicKey,
     stateBalance: Field,
     witness: MerkleMapWitness,
     channelDeltaBalance: Int64,
     channelNonce: Field,
-    channelBalanceSignature: Signature
-  ): Int64 {
+    channelBalanceSignature: Signature,
+    randomnessSignature: Signature,
+    encryptionCT1: Field,
+    encryptionCT2: Field,
+    encryptionGroup: Group,
+    executorPrivateKey: PrivateKey
+  ): [Int64, Field[]] {
     this.proveState(player, stateBalance, witness);
+    const oraclePublicKey = this.oraclePublicKey.get();
+    this.oraclePublicKey.assertEquals(oraclePublicKey);
+
     let channelBalanceProof = new ChannelBalanceProof(player, this.address);
     channelBalanceProof.deltaBalance = channelDeltaBalance;
     channelBalanceProof.nonce = channelNonce;
     channelBalanceProof.player.assertEquals(player);
     channelBalanceProof.verify(channelBalanceSignature).assertTrue();
 
-    // console.log(`In the method - before: ${channelBalanceProof.toString()}`);
-
     let trueBalance = channelBalanceProof.deltaBalance.add(
       Int64.fromField(stateBalance)
     );
 
-    // TODO: just using 100 as a random value for now, clean this up
-    const isValidFlip = Circuit.if(
-      trueBalance.isPositive(),
-      (() => trueBalance.toField().gt(100))(),
-      (() => Bool(false))()
+    const isValidFlip = trueBalance.toField().gt(Field(25));
+    isValidFlip.assertTrue('Balance is too low');
+
+    randomnessSignature
+      .verify(oraclePublicKey, [encryptionCT1, encryptionCT2])
+      .assertTrue();
+
+    const oracleRandomness = Encryption.decrypt(
+      {
+        publicKey: encryptionGroup,
+        cipherText: [encryptionCT1, encryptionCT2],
+      },
+      executorPrivateKey
     );
-    isValidFlip.assertTrue();
 
     const flipOutcome = Circuit.if(
-      Bool(false),
+      UInt64.fromFields(oracleRandomness).divMod(2).rest.equals(UInt64.zero),
       (() => Int64.fromField(Field(5)))(),
       (() => Int64.fromField(Field(5)).neg())()
     );
 
-    return flipOutcome;
+    return [flipOutcome, oracleRandomness];
```

Comment:
returning the recovered random number for UI purposes.  Might be nice to show the user.